plugin-server/src/worker/ingestion/event-pipeline/2-emitToBufferStep.ts

```diff
@@ -24,6 +24,28 @@ export async function emitToBufferStep(
         return runner.nextStep('processPersonsStep', event, personContainer)
     }
 
+    if (
+        process.env.DELAY_ALL_EVENTS_FOR_TEAMS === '*' ||
+        process.env.DELAY_ALL_EVENTS_FOR_TEAMS?.split(',').includes(event.team_id.toString())
+    ) {
+        // If we're delaying all events for this team, we don't want to run the
+        // complete pipeline on the event on the first pass, but rather just
+        // create a person and/or add new distinct_id, person_id pairs, then
+        // pass the event to the buffer for complete processing later.
+
+        // To communicate to the rest of the pipeline the difference of
+        // behavior we use the `fullyProcessEvent` flag on the runner, which
+        // should be available to all further steps.
+
+        // NOTE: we run in this step but we could have added a step, but I
+        // didn't want to add more steps then there already are.
+
+        // TODO: remove this conditional once we're confident that the new
+        // ingestion pipeline is working well for all teams.
+        runner.onlyUpdatePersonIdAssociations = true
+        return runner.nextStep('pluginsProcessEventStep', event, personContainer)
```

Comment:
It could be simpler if we did this instead:
2. plugins processing (so we always process plugins before buffer, could be useful for events that are dropped anyway to not go to the buffer & the person state here would be inconsistent for properties)
3. (potentially person ID handling) + emit to buffer
4. person handling (from buffer)

Reply:
This is a good idea but I think if I was using a plugin I'd prefer some lag + a better match on person than just running the plugin asap. This might be something where down the line we are reverse etl pushing to another service and to move a person from step 1 to 2 and having the right person would be really important there.

Reply:
I have [tests for two cases](https://github.com/PostHog/posthog/pull/13505/files#diff-4b66d551fbef355d8bf7d09eb82c707b0523fe08c2c8250fe41f08aa0b94df8dR119):

 1. plugins should be able to update attributes that person processing is dependent on
 2. if a plugin drops an event, it should not have any person side-effects

Happy to drop these requirements as desired.